sftp/sftp.go

```diff
@@ -0,0 +1,97 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SFTPManager interface abstracts the SFTP client
+type SFTPManager interface {
+	UploadFile(localFilePath, remoteDir string) error
+	DownloadFile(remoteFilePath, localDir string) error
+	DeleteFile(remoteFilePath string) error
```

Comment:
Given that this is just a file manager we can remove `File` at the end of each methods name.